plugins/parquet/parquetembed.cpp

```diff
@@ -0,0 +1,1982 @@
+{
+    bindUnicodeParam(chars, value, field, r_parquet);
+}
+
+/**
+ * @brief Calls the bind function for the data type of the value.
+ *
+ * @param len Length of QString
+ * @param value Data to be written to the parquet file.
+ * @param field Object with information about the current field.
+ */
+void ParquetRecordBinder::processQString(unsigned len, const char *value, const RtlFieldInfo *field)
+{
+    size32_t charCount;
+    rtlDataAttr text;
+    rtlQStrToStrX(charCount, text.refstr(), len, value);
+
+    bindStringParam(charCount, text.getstr(), field, r_parquet);
+}
+
+/**
+ * @brief Calls the bind function for the data type of the value.
+ *
+ * @param chars Number of chars in the value.
+ * @param value Data to be written to the parquet file.
+ * @param field Object with information about the current field.
+ */
+void ParquetRecordBinder::processUtf8(unsigned chars, const char *value, const RtlFieldInfo *field)
+{
+    bindUtf8Param(chars, value, field, r_parquet);
+}
+
+/**
+ * @brief Construct a new ParquetEmbedFunctionContext object
+ *
+ * @param _logctx Context logger for use with the ParquetRecordBinder ParquetDatasetBinder classes.
+ * @param options Pointer to the list of options that are passed into the Embed function.
+ * @param _flags Should be zero if the embedded script is ok.
+ */
+ParquetEmbedFunctionContext::ParquetEmbedFunctionContext(const IContextLogger &_logctx, const IThorActivityContext *activityCtx, const char *options, unsigned _flags)
+    : logctx(_logctx), m_scriptFlags(_flags)
+{
+    // Option Variables
+    const char *option = "";      // Read(read), Read Parition(readpartition), Write(write), Write Partition(writepartition)
+    const char *location = "";    // file name and location of where to write parquet file
+    const char *destination = ""; // file name and location of where to read parquet file from
+    __int64 rowsize = 40000;    // Size of the row groups when writing to parquet files
+    __int64 batchSize = 40000;  // Size of the batches when converting parquet columns to rows
+    // Iterate through user options and save them
+    StringArray inputOptions;
+    inputOptions.appendList(options, ",");
+    ForEachItemIn(idx, inputOptions)
+    {
+        const char *opt = inputOptions.item(idx);
+        const char *val = strchr(opt, '=');
+        if (val)
+        {
+            StringBuffer optName(val - opt, opt);
+            val++;
+            if (stricmp(optName, "option") == 0)
+                option = val;
+            else if (stricmp(optName, "location") == 0)
+                location = val;
+            else if (stricmp(optName, "destination") == 0)
+                destination = val;
+            else if (stricmp(optName, "MaxRowSize") == 0)
+                rowsize = atoi(val);
+            else if (stricmp(optName, "BatchSize") == 0)
+                batchSize = atoi(val);
+            else
+                failx("Unknown option %s", optName.str());
```

Comment:
It may be better to warn on an unknown option.  Failing catches mistyped options, but it also means if new options are added to ecl code the queries will fail on old builds.  Happy to leave as-is though.